### PR TITLE
feat(settings): add big file upload setup checks

### PR DESCRIPTION
**Pull request by skjnldsv**
https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/big_file_upload_configuration.html

We should have added that a long time ago :face_exhaling: 

image

### CODE REVIEW

File: apps/settings/lib/SetupChecks/PhpMaxFileSize.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	public function run(): SetupResult {
+		$upload_max_filesize = Util::computerFileSize(trim(ini_get('upload_max_filesize')));
+		$post_max_size = Util::computerFileSize(trim(ini_get('post_max_size')));
+		$max_input_time = (int)trim(ini_get('max_input_time'));
+		$max_execution_time = (int)trim(ini_get('max_execution_time'));
+
+		$warnings = [];
+		$recommendedSize = 16 * 1024 * 1024 * 1024;
```
Comment (szaimen) [inline apps/settings/lib/SetupChecks/PhpMaxFileSize.php:42]: Why 16GB and not 10GB for example? :D

**Review by szaimen — APPROVED** [review]
Nice idea!! 🤩

LGTM but didnt test

File: apps/settings/lib/SetupChecks/PhpMaxFileSize.php (hunk 1 above)
Comment (skjnldsv) [reply apps/settings/lib/SetupChecks/PhpMaxFileSize.php:42]: recommended value for years in our doc :shrug: 
I think we should stay consistent :)

Comment (szaimen) [reply apps/settings/lib/SetupChecks/PhpMaxFileSize.php:42]: Fine by me 👍

**Comment by szaimen** [comment]
Just an idea: can we also somehow check web server body limits and timeouts? (These are also important)

**Comment by skjnldsv** [comment]
> @skjnldsv Just an idea: can we also somehow check web server body limits and timeouts? (These are also important to test afaik)

Those variables are not accessible from php

**Review by come-nc — COMMENTED** [review]
You call trim on the result of ini_get, but ini_get can return false and trim takes a string? Is that not crashing when the ini settings are missing?

I’d feel safer if you use the iniGetWrapper used in the rest of the code.

File: apps/settings/lib/SetupChecks/PhpMaxFileSize.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		$recommendedTime = 3600;
+
+		// Check if the PHP upload limit is too low
+		if ($upload_max_filesize < $recommendedSize || $post_max_size < $recommendedSize) {
+			$warnings[] = $this->l10n->t('The PHP upload_max_filesize and/or post_max_size is too low. A file size of at least %1$s is recommended. Current value: %2$s.', [
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/settings/lib/SetupChecks/PhpMaxFileSize.php:47]: Sure :)

**Comment by skjnldsv** [comment]
> Is that not crashing when the ini settings are missing?

No, there is always a fallback by php defaults if the value is not defined

File: apps/settings/lib/SetupChecks/PhpMaxFileSize.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		}
+		if ($post_max_size < $recommendedSize) {
+			$warnings[] = $this->l10n->t('The PHP post_max_size is too low. A size of at least %1$s is recommended. Current value: %2$s.', [
+				Util::humanFileSize($recommendedSize),
+				Util::humanFileSize($post_max_size),
+			]);
+		}
+
+		// Check if the PHP execution time is too low
+		if ($max_input_time !== -1 && $max_execution_time !== -1) {
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/settings/lib/SetupChecks/PhpMaxFileSize.php:60]: yes, -1 = unlimited

[the remark this answers is not in the document]
Comment (skjnldsv) [reply apps/settings/lib/SetupChecks/PhpMaxFileSize.php:60]: ah no, I get it now, let me fix :facepalm:

**Review by come-nc — APPROVED** [review]

**Comment by nickvergessen** [comment]
Sorry but the values sounds extremely dangerous and like a security risk to me. Allowing 16 GB of input and a runtime of 1h per process is beyond acceptable and allows for a way too easy DOS attack.

Since we have chunking none of this should be needed. Chunk size of 512 MB is way over the top already, but I'm okay with keeping that.
The linked documentation mentions IE6 and other browsers which are long dead so the values might be from before chunking.

**Comment by ThibautPlg** [comment]
Hi,
I've just talked to the NC support about this subject and we came to the conclusion that chunking was used across Nextcloud (excluding public shares in old NC versions) so big values were not required anymore. 
Do I miss something here? 
Best regards,

**Comment by skjnldsv** [comment]
We're discussing this internally :)

**Pull request merged**